### PR TITLE
Add --force option to network rm subcommand 

### LINKED ISSUE #2382: Add a 'docker network rm --force' flag
## Desired Behavior
I need a `docker network rm --force` command that exits with success if the specified network doesn't exist.  This analogous to the `rm -f` shell command.
```
$ docker network rm --force nonexistent_network ; echo $?
0
```
## Current Behavior
Currently, the command exits with status 1 if no networks were deleted.
```
$ docker network rm nonexistent_network ; echo $?
Error: No such network: nonexistent_network
1
```

## Why I need this change
I am calling this command in a cleanup script.  I need to be able to run the script repeatedly, without creating any networks in between runs.  The script needs to stop and report an error if `docker network rm` cannot do what it needs to do.  For example, if it cannot connect to dockerd.  But if the networks are already deleted, then the script needs to continue.  With the current behavior of `docker network rm`, the script must perform its own check for networks before calling `docker network rm `.  This is extra code.  Alternatively, the script can silently ignore failure of the command.  This has serious drawbacks.

### PR DESCRIPTION
**Pull request by cavcrosby**
**- What I did**

Added a `-f` / `--force` option to the network's `rm` subcommand. What this option allows is for a user to remove a configured Docker network but return a zero status code even if said network does not exist (stdout will also still display the network name). Similar to Unix's `rm --force` command. This closes #2382.

**- How I did it**

I was inspired by the volume `rm` subcommand's `--force` option (at least on the client-side) so hence my implementation is similar to it. However, like #2678, my changes only affect the client and are currently not implemented for the daemon. 

**- How to verify it**

`docker network rm --force foo`

**- Description for the changelog**

Add --force option to network rm subcommand.

**- A picture of a cute animal (not mandatory but encouraged)**

image

### CODE REVIEW

**Comment by cavcrosby** [comment]
That's no bueno (or good), err, can I get a maintainer's thoughts?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks for working on this!

I left some suggestions inline, but was writing some of the changes while reviewing, so I'll open a pull request against your branch with those

**Comment by thaJeztah** [comment]
I opened https://github.com/cavcrosby/docker-cli/pull/1 (which, once merged, should appear in this PR); perhaps you can also squash the commits (no need to preserve my commit as a separate commit); happy to help doing so if you want!

File: cli/command/network/remove.go
Comment (cavcrosby) [reply cli/command/network/remove.go:32]: Yup, was a copypasta from the `docker volume rm`.

**Comment by cavcrosby** [comment]
Squashed your commit, and closed out the PR on my fork. Should be good to go but let me know what you think.

**Review by albers — APPROVED** [review]
Completion LGTM, thanks.

**Review by thaJeztah — COMMENTED** [review]
changes LGTM, thanks!

I think we can combine the 5 commits into one, as they're all related (and some of the later ones touch up previous ones); let me do a quick squash and push to the PR branch

**Review by thaJeztah — APPROVED** [review]

**Comment by thaJeztah** [comment]
All green; let's get this one in

Thanks!

**Pull request merged**
